baremetal_network_provisioning/ml2/extensions/switch.py

```diff
@@ -12,29 +12,225 @@
+        'vendor': {'allow_post': True, 'allow_put': False,
+                   'validate': {'type:string': None},
+                   'is_vidible': True}
+    },
 }
 
 
+def validator_func(data, valid_values=None):
+    """Validate the access parameters."""
+    if not data:
+        # Access parameters must be provided.
+        msg = _("Cannot create a switch from the given input.")
+        return msg
+    if type(data) is not dict:
+        msg = _("Given details is not in the form of a dictionary.")
+        return msg
+
+attributes.validators['type:access_dict'] = validator_func
+
+
 class BNPSwitchController(wsgi.Controller):
 
+    @property
+    def _dbcontext(self):
+        return context.get_admin_context()
+
     def index(self, request):
-        pass
+        switches = db.get_all_bnp_phys_switches(self._dbcontext)
+        switches_dict = {
+            'switches': [switch.__dict__ for switch in switches]}
+        return switches_dict
 
     def show(self, request, id):
-        pass
+        switch = db.get_bnp_phys_switch(self._dbcontext, id)
+        if not switch:
+            raise exc.NotFound(
+                resource="switch %s" % (id))
+        switch_dict = switch.__dict__
+        switch_dict['ports'] = []
+        # Get a list of Ironic ports
+        return switch_dict
 
     def delete(self, request, id):
-        pass
+        switch = db.get_bnp_phys_switch(self._dbcontext, id)
+        if not switch:
+            raise exc.NotFound(
+                resource="switch %s" % (id))
+        db.delete_bnp_phys_switch(self._dbcontext, id)
 
     def create(self, request):
-        pass
+        try:
+            body = request.json_body
+        except json_scanner.JSONDecodeError:
+            raise exc.BadRequest(
+                resource="switch",
+                reason="invalid JSON body")
+        try:
+            body = body.pop("switch")
+        except KeyError:
+            raise exc.BadRequest(
+                resource="switch",
+                reason="'switch' not found in request body")
+        keys = body.keys()
+        key_list = ['ip_address', 'vendor',
+                    'access_protocol', 'access_parameters']
+        for key in key_list:
+            if key not in keys:
+                raise exc.BadRequest(
+                    resource="switch",
+                    reason="'Key %s' not found in request body" % key)
+        if body['access_protocol'].lower() not in const.SUPPORTED_PROTOCOLS:
+            raise exc.BadRequest(
+                resource="switch",
+                reason="'Protocol %s' is not supported" % body[
+                    'access_protocol'])
+        access_parameters = body.pop("access_parameters")
+        if body['access_protocol'].lower() == 'snmpv3':
+            self._validate_snmpv3_parameters(access_parameters)
+        else:
+            self._validate_snmp_parameters(access_parameters)
+        switch_dict = self._create_switch_dict()
```

Comment:
why create_dict & update_dict ? cant we handle in create_dict itself by passing the values being supplied by discovery driver?